### PR TITLE
Docker rabbitmq setup works better out of the box

### PR DESCRIPTION
**Pull request by c-knowles**
- Introduce docker-compose file
- Correct certificate file names in spec config
- Push port expose to docker network to avoid conflict and ease setup,
also allow guest user to connect to rabbitmq "remotely" as docker won't
be on the loopback interface any more
- Allow peers to verify inside tests by setting up an alt name in the
certificate (currently all tests are set to not verify)

I did this work as part of getting setup for https://github.com/ruby-amqp/bunny/issues/440. It was more difficult than I had hoped to get setup so this hopefully helps with that.

It may also be of interest to use the docker compose setup in the [CI](https://docs.travis-ci.com/user/docker/#Using-Docker-Compose) then we could get rid of `unless ENV["CI"]` and have some of the tests using `verify_peer` on.

### CODE REVIEW

**Comment by c-knowles** [comment]
I did this work as part of getting setup for https://github.com/ruby-amqp/bunny/issues/440. It was more difficult than I had hoped to get setup so this hopefully helps with that.

File: spec/tls/generate-server-cert.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+DIRNAME=$(dirname "$0")
+
+openssl req -sha256 -new -key $DIRNAME/server_key.pem -out $DIRNAME/server.csr -subj "/CN=mercurio" -config $DIRNAME/server-openssl.cnf
```
Comment (michaelklishin) [inline spec/tls/generate-server-cert.sh]: You can potentially use [tls-gen](https://github.com/michaelklishin/tls-gen).

**Comment by c-knowles** [comment]
It may also be of interest to use the docker setup in the CI then we could get rid of `unless ENV["CI"]` and have some of the tests using `verify_peer` on.

**Comment by c-knowles** [comment]
I wanted to know if `spec/config/rabbitmq.config` is used from CI? I could not spot a usage but I left the file references as relative paths. If we can change that file freely then the `docker-compose.yml` also gets a bit neater.

**Comment by michaelklishin** [comment]
No, Travis starts a RabbitMQ node managed via `sudo service`.

**Comment by michaelklishin** [comment]
This doesn't have any side effects on the "local" workflow, so merging. We can make further corrections down the road.

**Pull request merged**